lychee-lib/src/extract/html/html5gum.rs

```diff
@@ -147,7 +150,15 @@ impl LinkExtractor {
             if attr == "rel" && value.contains("nofollow") {
                 self.current_element_nofollow = true;
             }
-            if self.current_element_nofollow {
+
+            // Ignore links with rel=preconnect
+            // Other than prefetch and preload, preconnect only makes
+            // a DNS lookup, so we don't want to extract those links.
+            if attr == "rel" && value.contains("preconnect") {
+                self.current_element_preconnect = true;
```

Comment:
you are not resetting this attribute to false anywhere, i haven't checked the testfailures yet

Reply:
Ouch. Fixed.